src/lib.rs

```
@@ -238,12 +241,8 @@ impl<'a> fmt::Write for PlatformLogWriter<'a> {
 ///
 /// It is ok to call this at the activity creation, and it will be
 /// repeatedly called on every lifecycle restart (i.e. screen rotation).
-pub fn init_once(log_level: LogLevel) {
-    match log::set_logger(|max_log_level| {
-        max_log_level.set(log_level.to_log_level_filter());
-        return Box::new(AndroidLogger::default());
-    }) {
-        Err(e) => debug!("{}", e),
-        _ => (),
-    }
+pub fn init_once(log_level: Level) {
+    log::set_max_level(log_level.to_level_filter());
+    log::set_logger(&AndroidLogger).unwrap();
```

It is library, so may be move descion do `panic!` or not,
out of library and return from `init_once` `Result<(), Something>`?

---

To add more details, this function call may run on each android lifecycle change (i.e. screen rotation), because android tears up everything _except_ native libs. So, if written as-is, it would panic the second time it is called. Hence the name "init once", it is meant to ignore this second initialization.

Another function named "init" that returns the result may be better, but this can be done in separate PR if needed.

---

I changed it to a panic because it felt _really_ weird to me to ignore the error otherwise (as trying to log without a logger does nothing useful).

I can return the error instead if that is wanted.

---

Yes, ignoring an error feels weird inside a lib. However, if this function panics, there is no one to log the panic; even worse, if there is no global panic catcher, we are suddenly unwinding into android guts.

Therefore for initial version, I ignored this error, and _to submit this PR quickly, you can ignore it too :)_.

However, if we want to do this properly (and return the Error), we need to take care to only initialize logger successfully _once_. Basically, we want to ignore "logger is already initialized" error from log::set_logger, but not ignore any other error.

You may try to do it if you want :)

---

Sure. That sounds like the ideal. I'll try and get some time within the next couple days to wrap that up.